### PR TITLE
Computed the banded matrix for each column of stacked mesh

### PR DESCRIPTION
**Pull request by jkozdon**

### CODE REVIEW

**Comment by jkozdon** [comment]
bors try

**Comment by lcw** [comment]
This is looking good.  Do we want to pass in a function to build the matrix of?  Some function that can be provided by the ode solver.

File: src/DGmethods/DGmodel.jl [hunk 1]
```diff
@@ -244,3 +246,112 @@ function nodal_update_aux!(f!, dg::DGModel, m::BalanceLaw, Q::MPIStateArray,
...
+function MPIStateArrays.MPIStateArray(dg::DGModel, commtag=888)
+  bl = dg.balancelaw
+  grid = dg.grid
+
+  state = create_state(bl, grid, commtag)
+
+  return state
+end
+
+function banded_matrix(dg::DGModel, Q = MPIStateArray(dg),
```
Comment (simonbyrne) [inline src/DGmethods/DGmodel.jl]: Can you add a docstring?

**Comment by jkozdon** [comment]
bors try

File: src/DGmethods/DGmodel.jl (hunk 1 above)
Comment (jkozdon) [reply src/DGmethods/DGmodel.jl]: done.

**Comment by jkozdon** [comment]
> This is looking good. Do we want to pass in a function to build the matrix of? Some function that can be provided by the ode solver.

Added a simple interface for this.

**Comment by jkozdon** [comment]
bors try-

**Comment by jkozdon** [comment]
bors try

**Comment by jkozdon** [comment]
There seem to be some issues with `Float32` and the linear atmos model. Guessing the conditioning is poor, since we have order `1` and `1e7` numbers in the matrix.

Not sure if the test setup can be modified to improve this.

**Comment by jkozdon** [comment]
@lcw proposed closing this PR and merging with #518 as needed

**Pull request closed without merging**